contracts/exchange-forwarder/test/forwarder.ts

```diff
@@ -186,244 +195,89 @@ describe(ContractName.Forwarder, () => {
-                    zrxAssetData,
                     wethAssetData,
                 ) as any) as sendTransactionResult,
                 RevertReason.UnregisteredAssetProxy,
             );
         });
     });
     describe('marketSellOrdersWithEth without extra fees', () => {
-        it('should fill a single order', async () => {
-            const ordersWithoutFee = [orderWithoutFee];
-            const feeOrders: SignedOrder[] = [];
-            const ethValue = orderWithoutFee.takerAssetAmount.dividedToIntegerBy(2);
-
-            tx = await forwarderWrapper.marketSellOrdersWithEthAsync(ordersWithoutFee, feeOrders, {
-                value: ethValue,
-                from: takerAddress,
-            });
-            const takerEthBalanceAfter = await web3Wrapper.getBalanceInWeiAsync(takerAddress);
-            const forwarderEthBalance = await web3Wrapper.getBalanceInWeiAsync(forwarderContract.address);
-            const newBalances = await erc20Wrapper.getBalancesAsync();
-
-            const primaryTakerAssetFillAmount = ForwarderWrapper.getPercentageOfValue(
-                ethValue,
-                MAX_WETH_FILL_PERCENTAGE,
-            );
-            const makerAssetFillAmount = primaryTakerAssetFillAmount
-                .times(orderWithoutFee.makerAssetAmount)
-                .dividedToIntegerBy(orderWithoutFee.takerAssetAmount);
-            const totalEthSpent = primaryTakerAssetFillAmount.plus(gasPrice.times(tx.gasUsed));
-
-            expect(takerEthBalanceAfter).to.be.bignumber.equal(takerEthBalanceBefore.minus(totalEthSpent));
-            expect(newBalances[makerAddress][defaultMakerAssetAddress]).to.be.bignumber.equal(
-                erc20Balances[makerAddress][defaultMakerAssetAddress].minus(makerAssetFillAmount),
-            );
-            expect(newBalances[takerAddress][defaultMakerAssetAddress]).to.be.bignumber.equal(
-                erc20Balances[takerAddress][defaultMakerAssetAddress].plus(makerAssetFillAmount),
-            );
-            expect(newBalances[makerAddress][weth.address]).to.be.bignumber.equal(
-                erc20Balances[makerAddress][weth.address].plus(primaryTakerAssetFillAmount),
-            );
-            expect(newBalances[forwarderContract.address][weth.address]).to.be.bignumber.equal(constants.ZERO_AMOUNT);
-            expect(newBalances[forwarderContract.address][defaultMakerAssetAddress]).to.be.bignumber.equal(
-                constants.ZERO_AMOUNT,
-            );
-            expect(forwarderEthBalance).to.be.bignumber.equal(constants.ZERO_AMOUNT);
+        it('should fill a single order without a taker fee', async () => {
+            await forwarderTestFactory.marketSellTestAsync([orderWithoutFee], new BigNumber(0.5), erc20Token);
         });
-        it('should fill multiple orders', async () => {
+        it('should fill multiple orders without taker fees', async () => {
             const secondOrderWithoutFee = await orderFactory.newSignedOrderAsync();
```

Comment:
I would prefer if these orders had different maker/taker amounts. There are some other places where we generate identical orders, as well.